### PR TITLE
FIX#98 resetting ConfigSpace random number generator

### PR DESCRIPTION
**Pull request by KEggensperger**
This PR draws a random integer with MAXINT as its maximum to reset the rng in the ConfigSpace object in the SMAC facade, which is used by all other facades.

It also replaces '1234567980' with MAXINT as the maximum random integer to seed the random forest.

### CODE REVIEW

File: smac/facade/smac_facade.py [hunk 1]
```diff
@@ -93,6 +94,9 @@ def __init__(self,
         # initial random number generator
         num_run, rng = self._get_rng(rng=rng)
 
+        # reset random number generator in config space
+        scenario.cs.random = np.random.RandomState(rng.randint(MAXINT))
```
Comment (mfeurer) [inline smac/facade/smac_facade.py]: I think you should either do `scenario.cs.seed(rng.randint(MAXINT))` or `scenario.cs.random = rng`, but not create a new random state here.

Comment (KEggensperger) [inline smac/facade/smac_facade.py]: fixed

File: smac/initial_design/single_config_initial_design.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+        status, cost, runtime, additional_info = self.tae_runner.start(
+            initial_incumbent,
+            instance=rand_inst,
+            cutoff=self.scenario.cutoff,
+            seed=initial_seed,
+            instance_specific=self.scenario.instance_specific.get(rand_inst, "0"))
+
+        if status in [StatusType.CRASHED or StatusType.ABORT]:
+            self.logger.critical("First run crashed -- Abort")
```
Comment (mfeurer) [inline smac/initial_design/single_config_initial_design.py]: You could fix #85 here if you want ;)

File: test/test_facade/test_roar_facade.py [hunk 3]
```diff
@@ -22,3 +24,41 @@ def test_inject_stats_and_runhistory_object_to_TAE(self):
         ROAR(tae_runner=ta, scenario=self.scenario)
         self.assertIsInstance(ta.stats, Stats)
         self.assertIsInstance(ta.runhistory, RunHistory)
+
+    def test_check_random_states(self):
```
Comment (mfeurer) [inline test/test_facade/test_roar_facade.py]: I think these tests suffice in the smac facade and don't have to be duplicated.

File: smac/scenario/scenario.py [hunk 4]
```diff
@@ -190,6 +189,10 @@ def _add_arguments(self):
                           dest='train_insts')
         self.add_argument(name='test_instances', default=[[None]], help=None,
                           dest='test_insts')
+        self.add_argument(name='initial_incumbent', default="DEFAULT",
+                          help=None, dest='initial_incumbent',
+                          callback=lambda arg: str.upper(arg.strip()) if
+                          str.upper(arg.strip()) in ['DEFAULT', 'RANDOM'] else None)
```
Comment (mfeurer) [inline smac/scenario/scenario.py]: Hm, although we should have a choice here, a hotfix would be raising an exception in the else statement. Would that make sense?

File: smac/initial_design/single_config_initial_design.py (hunk 2 above)
Comment (KEggensperger) [inline smac/initial_design/single_config_initial_design.py]: Should be already fixed

File: test/test_facade/test_roar_facade.py (hunk 3 above)
Comment (KEggensperger) [inline test/test_facade/test_roar_facade.py]: agree

File: smac/scenario/scenario.py (hunk 4 above)
Comment (KEggensperger) [inline smac/scenario/scenario.py]: agree, but that is not a valid statement

**Pull request merged**